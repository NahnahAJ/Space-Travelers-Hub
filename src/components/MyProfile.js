// Render a list of all joined missions (use filter()) on the "My profile" page.
import React from 'react';
import { useSelector } from 'react-redux';
import { Table } from 'react-bootstrap';

export default function MyProfile() {
  const res = (state) => {
    if (state.reserved === true) {
      return { ...state, reserved: true };
    }
    return state.reserved;
  };

  const currentState = useSelector((state) => state.mission.missions);
  const newArray = currentState.filter(res);

  return (
    <div>
      <h4>My Missions</h4>
      <Table bordered hover>
        <tbody>
          {
          newArray.map((res) => (
            <tr key={res.missionId}>
              <td>{res.missionName}</td>
            </tr>
          ))
        }
        </tbody>
      </Table>
    </div>
  );
}
