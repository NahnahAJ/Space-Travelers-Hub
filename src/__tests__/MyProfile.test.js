import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

test('renders the myProflie component snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
      <Provider store={store}>
        <MyProfile />
      </Provider>
    </BrowserRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
