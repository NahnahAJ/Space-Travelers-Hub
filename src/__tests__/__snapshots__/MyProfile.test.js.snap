// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the myProflie component snapshot 1`] = `
<DocumentFragment>
  <div
    class="myProfile"
  >
    <div
      class="myMissions"
    >
      <h4>
        My Missions
      </h4>
      <table
        class="table table-bordered table-hover"
      >
        <tbody />
      </table>
    </div>
    <div
      class="myRockets"
    >
      <h4>
        My Rockets
      </h4>
      <table
        class="table table-bordered table-hover"
      >
        <tbody />
      </table>
    </div>
  </div>
</DocumentFragment>
`;
